datafusion/common/src/scalar.rs

```diff
@@ -1705,6 +1705,29 @@ impl ScalarValue {
         })
     }
 
+    /// Create an one value in the given type.
+    pub fn new_one(datatype: &DataType) -> Result<ScalarValue> {
+        assert!(datatype.is_primitive());
+        Ok(match datatype {
+            DataType::Boolean => ScalarValue::Boolean(Some(true)),
```

Comment:
Boolean doesn't make sense to me as it isn't numeric. I think Boolean should also return an error